spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/util/SerializationStrategy.java

```diff
@@ -0,0 +1,102 @@
+package org.springframework.security.oauth2.common.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Serialization strategy based on {@link SaferObjectInputStream}.
+ */
+public class SerializationStrategy {
```

Comment:
Let's make this an `interface` and name the whitelisted implementation `WhitelistedSerializationStrategy` or `DefaultSerializationStrategy`

Reply:
Okay, no problem.